### PR TITLE
feat: add interval field to add health form

### LINKED ISSUE #809: Settings - > Health - Add interval field

### PR DESCRIPTION
**Pull request by mainawycliffe**
closes #809

### CODE REVIEW

File: src/components/AutoCompleteDropdown/AutoCompleteDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    );
+  }
+
+  return (
+    <Select
+      className="w-full"
+      options={options}
+      onChange={(value) => onChange(value?.value ?? undefined)}
+      defaultInputValue={value}
+      onInputChange={(event) => console.log(event)}
```
Comment (moshloop) [inline src/components/AutoCompleteDropdown/AutoCompleteDropdown.tsx:32]:
```suggestion
```

File: src/components/SchemaResourcePage/SchemaResource.tsx [hunk 2]
```diff
@@ -40,14 +41,28 @@ export function SchemaResource({
...
+        });
+      } else {
+        await updateResource(resourceInfo, {
+          ...resource,
+          ...props,
+          updated_at: new Date().toISOString()
+        });
+      }
     } catch (ex: any) {
+      console.error(ex);
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResource.tsx:65]:
```suggestion
```

File: src/components/SchemaResourcePage/SchemaResource.tsx [hunk 3]
```diff
@@ -40,14 +41,28 @@ export function SchemaResource({
...
-      });
+      if (resourceInfo.table === "canaries") {
+        const { interval, spec, ...dataWithoutInterval } = props;
+        const specWithInterval = {
+          ...(spec as {}),
+          ...(interval ? { interval: interval } : {})
+        };
+        await updateResource(resourceInfo, {
+          ...dataWithoutInterval,
+          updated_at: new Date().toISOString(),
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResource.tsx:54]: We should never be populating the updated_at client side.

File: src/components/SchemaResourcePage/SchemaResourceEdit.tsx [hunk 4]
```diff
@@ -58,10 +67,21 @@ export function SchemaResourceEdit({
...
+      spec: spec ? load(spec as string) : undefined,
+      name,
+      labels,
+      namespace,
+      icon
+    },
     identity
   );
+
+  console.log("defaultValues", defaultValues);
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResourceEdit.tsx:83]:
```suggestion
```

**Comment by moshloop** [comment]
When I enter a value and hit enter, it navigates away, 
I actually think we should be updating the schedule field, not the interval: https://github.com/flanksource/duty/blob/main/schema/checks.hcl#L24

**Comment by mainawycliffe** [comment]
@moshloop Yeah, Yash told me that earlier, which was the reason the error was happening, I will make the changes.

**Comment by mainawycliffe** [comment]
@moshloop quick question, in the schema, the field is schedule, do we name it to schedule in the UI?

**Comment by mainawycliffe** [comment]
@moshloop and another question, do we want to change the values to be something like the others e.g: `@every 30s` instead of just 30?

**Comment by mainawycliffe** [comment]
@moshloop this is good to go

File: src/components/SchemaResourcePage/SchemaResourceEdit.tsx [hunk 5]
```diff
@@ -301,6 +324,67 @@ export function SchemaResourceEdit({
...
+                                  </label>
+                                  <AutoCompleteDropdown
+                                    onChange={onChange}
+                                    value={value}
+                                    isDisabled={!!source || disabled || !edit}
+                                    options={[
+                                      {
+                                        label: "@every 10s",
+                                        value: "@every 10s"
+                                      },
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResourceEdit.tsx:347]:
```suggestion
```

**Pull request merged**